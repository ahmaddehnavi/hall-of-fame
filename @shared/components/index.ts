export * from './layout/Col'
export * from './layout/Row'
export * from './screen/Screen'
export * from './backhandler/MultiBackHandler'
export * from './touchable/Touchable'
export * from './button/Button'
export * from './list/FlatList'
export * from './empty/EmptyView'
export * from './loading/LoadingView'
export * from './error/ErrorView'
export * from './async-view/AsyncView'