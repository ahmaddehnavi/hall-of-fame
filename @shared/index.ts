export * from './services'
export * from './utils'
export * from './components'
export * from './resource/Resource'
