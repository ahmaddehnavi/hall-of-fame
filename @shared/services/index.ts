export * from './base'
export * from './navigation'
export * from './theme'

