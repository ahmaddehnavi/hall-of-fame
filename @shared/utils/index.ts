export * from './di/DIUtil'
export * from './async/AsyncUtil'