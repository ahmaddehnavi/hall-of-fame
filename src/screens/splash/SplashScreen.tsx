import {PXInjectedNavigationServiceProps, PXScreen} from '@shared';
import {inject} from 'mobx-react';
import React from 'react';
import {ActivityIndicator, StyleSheet} from 'react-native';
import {IntroScreen} from '../intro/IntroScreen';

@inject('$navigation')
export class SplashScreen extends React.Component<PXInjectedNavigationServiceProps> {
    static readonly ROUTE_NAME = 'SplashScreen';

    componentDidMount() {
        setTimeout(() => {
            IntroScreen.start(this.props.$navigation)
        },1000);
    }

    render() {
        return (
            <PXScreen style={styles.container}>
                <ActivityIndicator/>
            </PXScreen>
        )
    }
}


const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'center',
    }
});