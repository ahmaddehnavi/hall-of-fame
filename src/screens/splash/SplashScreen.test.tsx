import {PXNavigationService} from '@shared';
import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {SplashScreen} from './SplashScreen';

it('renders correctly', () => {
    let tree = renderer.create(
    <SplashScreen
       $navigation = {new PXNavigationService()}
    />
)
    ;
    expect(tree).toMatchSnapshot();
});

