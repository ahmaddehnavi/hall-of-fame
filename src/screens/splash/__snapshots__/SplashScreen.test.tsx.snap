// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Screen test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
  <Text>
    Loading...
  </Text>
</View>
`;
